libraries/src/Mail/MailTemplate.php

```diff
@@ -485,29 +486,20 @@ public static function deleteTemplate($key)
 	 *
 	 * @since   4.0.0
 	 */
-	protected function setAttachmentName($file, $name)
+	protected function getAttachmentName($file, $name)
 	{
-		// If no name is given use the basename of the file to be attached
+		// If no name is given, do not process it further
 		if (!trim($name))
 		{
-			return basename($file);
+			return '';
```

Comment:
Have you really tested that it falls back to the file name when the email is sent and the name is empty?

Reply:
Haven't tested but I use the addAttachment method in my extensions for years. If second parameter is empty, it will  use the original name. That second parameter is always optional and default to empty string, see https://github.com/joomla/joomla-cms/blob/4.0-dev/libraries/src/Mail/Mail.php#L419